### PR TITLE
Optimizations on Toast Component

### PR DESCRIPTION
**Pull request by elena-dumitrescu**

### CODE REVIEW

File: src/components/feedback/Toast/useToast.ts [hunk 1]
```diff
@@ -1,64 +1,53 @@
...
   return useCallback(
     /**
      *
-     * @param {ToastContent} message The content to be displayed
+     * @param {ToastContent} message The content to be displayed. It can be a string or a React component.
      * @param {('success'|'info'|'warning'|'error')} variant The type of the toast
      * @param {ToastOptions} options Additional options passed to the toast
      */
-    (message: ToastContent, variant?: ToastifyTypeOptions, { transitionType, ...restOptions } = {} as ToastOptions) => {
+    (message: ToastContent, variant?: ToastifyTypeOptions, { transitionType, actions, textProps, ...restOptions } = {} as ToastContainerProps) => {
```
Comment (alexandra-c) [inline src/components/feedback/Toast/useToast.ts:21]: options are of type ToastOptions not ToastContainerProps.
revert the wrong casting done here, recreate the ToastOptions type and extend it with the new properties you need.

File: src/components/feedback/Toast/useToast.ts [hunk 2]
```diff
@@ -1,64 +1,53 @@
 import { useCallback } from 'react'
 import {
   toast,
-  Slide,
-  Bounce,
-  Flip,
-  Zoom,
   ToastContent,
-  ToastOptions as ToastifyToastOptions,
+  ToastOptions,
```
Comment (alexandra-c) [inline src/components/feedback/Toast/useToast.ts:5]: rename to ToastOptionsBase

File: src/components/feedback/Toast/types.ts [hunk 3]
```diff
@@ -1,11 +1,29 @@
 // Copyright (c) TotalSoft.
 // This source code is licensed under the MIT license.
-import { ToastContainerProps as ReactToastifyProps } from 'react-toastify'
+import { TypographyProps } from 'components/dataDisplay/Typography'
+import { always, cond, equals, T } from 'ramda'
+import { Bounce, Flip, ToastContainerProps as ReactToastifyProps, Slide, Zoom } from 'react-toastify'
```
Comment (alexandra-c) [inline src/components/feedback/Toast/types.ts:5]: rename ToastContainerPropsBase

File: src/components/feedback/Toast/ToastStyles.ts [hunk 4]
```diff
@@ -11,47 +12,57 @@ export const classes = {
...
+      '--toastify-icon-color-error': theme.palette.error.main,
+      '--toastify-color-progress-error': theme.palette.error.main
+    },
+    [`& .${classes.warning}`]: {
+      '--toastify-color-warning': theme.palette.warning.main,
+      '--toastify-text-color-warning': theme.palette.warning.contrastText,
+      '--toastify-icon-color-warning': theme.palette.warning.main,
+      '--toastify-color-progress-warning': theme.palette.warning.main
+    },
+    ['.Toastify__close-button']: {
```
Comment (alexandra-c) [inline src/components/feedback/Toast/ToastStyles.ts:50]: cred ca nu mai avem nevoie de style-urile astea, pt ca folosim butonul nostru pt close acum mereu

File: src/components/feedback/Toast/ToastContainer.tsx [hunk 5]
```diff
@@ -13,15 +13,22 @@ const ToastContainer: React.FC<ToastContainerProps> = ({
   position = 'top-center',
   newestOnTop = true,
   transitionType = 'Slide',
+  closeOnClick = true,
+  closeButton = false,
+  icon = false,
   limit = 5,
+  textSize = 'large',
```
Comment (alexandra-c) [inline src/components/feedback/Toast/ToastContainer.tsx:20]: small

File: src/components/feedback/Toast/ToastStyles.ts (hunk 4 above)
Comment (elena-dumitrescu) [reply src/components/feedback/Toast/ToastStyles.ts:50]: mai avem nevoie pentru usePromise

File: src/components/feedback/Toast/usePromiseToast.ts [hunk 6]
```diff
@@ -10,13 +17,21 @@ const usePromiseToast = () => {
      * @param {(String|Object)} pending The message to be shown while promise in pending or the entire object with all the configurations
      * @param {(String|Object)} success The message to be shown when promise completed successfully or the entire object with all the configurations
      * @param {(String|Object)} error The message to be shown when promise was rejected or the entire object with all the configurations
+     * @param {ToastOptions} options Additional options passed to the toast
      */
-    (promise: Promise<unknown>, pending: string | object, success: string | object, error: string | object) => {
+    (promise: Promise<unknown>, pending: string | object, success: string | object, error: string | object, { transitionType, ...restOptions } = {} as ToastOptions) => {
```
Comment (andra-sava) [inline src/components/feedback/Toast/usePromiseToast.ts:22]: Here, in place of "as"... I thik there should be ":". To enforce the type.. not to cast the type you are receiving.

File: src/components/feedback/Toast/useToast.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     *
+     * @param {ToastContent} message The content to be displayed. It can be a string or a React component.
+     * @param {('success'|'info'|'warning'|'error')} variant The type of the toast
+     * @param {ToastOptions} options Additional options passed to the toast
+     */
+    (
+      message: ToastContent,
+      variant?: ToastifyTypeOptions,
+      { transitionType, actions, ...restOptions } = {} as ToastOptions
```
Comment (andra-sava) [inline src/components/feedback/Toast/useToast.tsx:22]: Also here.

File: src/stories/feedback/Toast/ActionsPreview.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) TotalSoft.
+// This source code is licensed under the MIT license.
+
+import React from 'react'
+import { Button, Typography, useToast } from 'components'
+import { emptyFunction } from 'components/utils/constants'
```
Comment (andra-sava) [inline src/stories/feedback/Toast/ActionsPreview.tsx:6]: This import also.

File: src/components/feedback/Toast/usePromiseToast.ts (hunk 6 above)
Comment (elena-dumitrescu) [reply src/components/feedback/Toast/usePromiseToast.ts:22]: won't do

**Review by andra-sava — APPROVED** [review]

**Pull request merged**